### PR TITLE
Update testcontainers

### PR DESCRIPTION
**Pull request by KyleAure**
Accomplishes the following goals: 

- Ensures `.docker/config.json` is updated with valid JSON in the expected format for docker-desktop to prevent additional parsing
- Replace `componenttest.containers.ExternalDockerClientStrategy` with `org.testcontainers.dockerclient.EnvironmentAndSystemPropertyClientProviderStrategy`
- Upgrade testcontainers to 1.18.3
- Upgrade docker-java to 3.3.1

consider waiting to merge this PR until after https://github.com/OpenLiberty/open-liberty/pull/25624 to prevent merge conflicts

### CODE REVIEW

**Comment by KyleAure** [comment]
#build

File: dev/fattest.simplicity/src/componenttest/containers/ArtifactoryRegistry.java [hunk 1]
```diff
@@ -121,59 +130,56 @@ private static File generateDockerConfig(String registry, String authToken, File
...
-        writeFile(configFile, contents);
+
+        //If existing config contains correct registry and auth token combination, return original file.
+        try {
+            if (root.get("auths").get(registry).get("auth").textValue().equals(authToken)) {
+                Log.info(c, m, "Config already contains the correct auth token");
+                return configFile;
+            }
+        } catch (Exception ignore) {
+            //assume config did not contain authToken
```
Comment (mswatosh) [inline dev/fattest.simplicity/src/componenttest/containers/ArtifactoryRegistry.java]: Should we still log the exception here even if it's ignored, for potential debugging?

**Comment by KyleAure** [comment]
#build

**Review by mswatosh — DISMISSED** [review]

**Comment by KyleAure** [comment]
#build

**Review by mswatosh — DISMISSED** [review]

**Comment by KyleAure** [comment]
#libby #build

**Comment by KyleAure** [comment]
The build failed with existing defects. 
Also, a child build never finished.  
Requested re-build here: https://wasrtc.hursley.ibm.com:9443/jazz/web/projects/WS-CD#action=com.ibm.team.build.viewDefinition&id=_ZquxgBOhEeeNhcSO7gcIdw

Target locations of links might be accessible only to IBM employees.

**Review by mswatosh — DISMISSED** [review]

**Comment by KyleAure** [comment]
#build - doing a new build to see if we hit the same child build issue.

**Comment by KyleAure** [comment]
#build

**Comment by KyleAure** [comment]
#build had to resolve merge conflicts 😟

**Comment by KyleAure** [comment]
Build failed with unrelated issue.  Ready to merge.

**Comment by KyleAure** [comment]
#libby

**Review by mswatosh — APPROVED** [review]

**Pull request merged**